front/pages/api/registry/[type]/lookup.ts

```diff
@@ -86,13 +87,13 @@ async function handler(
             res.status(404).end();
             return;
           }
-
-          const dataSource = await DataSource.findOne({
-            where: {
-              workspaceId: owner.id,
-              name: req.query.data_source_id,
-            },
-          });
+          const auth = await Authenticator.internalBuilderForWorkspace(
```

Comment:
Here given that it's an API route we should be able to get an authenticator from the key

Reply:
no key for registry calls from the core (only a special secret)